sound/soc/sof/sof-spi-dev.c

```diff
@@ -121,6 +121,7 @@ static int sof_spi_probe(struct spi_device *spi)
 	mach->sof_fw_filename = desc->nocodec_fw_filename;
 	mach->sof_tplg_filename = desc->nocodec_tplg_filename;
 	mach->asoc_plat_name = "sof-platform";
+	mach->mach_params.platform = "sof-audio";
```

Comment:
@plbossart would it make sense to do it in sof_probe() before the machine driver is registered instead of doing it individually?

Reply:
@ranj063 yes I thought about this but no: the core doesn't know anything about mach_params or ACPI, the parameter used for platform_device_register_data is a opaque void *. In other words, you can only set this at a higher level. You may view it as not elegant but it preserves the independence of the core from Intel/ACPI-specific stuff.